### PR TITLE
build(ci): Add breeze build and test job

### PR DESCRIPTION
**Pull request by dreveman**

### CODE REVIEW

**Comment by dreveman** [comment]
@Yuhta This fixes a number of build issues in breeze and introduces CI jobs to build and run all breeze tests on openmp and cuda. I can split this up into multiple PRs if we're not ready to add GPU jobs to CI at this time?

**Comment by dreveman** [comment]
@jon-rv @zytyz fyi

**Comment by assignUser** [comment]
Can we move this into it's own workflow so we can path filer on breeze so we only run it when necessary?

File: .github/workflows/linux-build-base.yml [hunk 1]
```diff
@@ -176,3 +176,78 @@ jobs:
       - name: Run Tests
         run: |
           cd _build/debug && ctest -j 8 --output-on-failure --no-tests=error
+
+  breeze-ubuntu-debug:
+    runs-on: 8-core-ubuntu
```
Comment (assignUser) [inline .github/workflows/linux-build-base.yml:181]: The build time is quite fast so we don't need the more expensive runner imo.
```suggestion
    runs-on: ubuntu-22.04
```

**Comment by dreveman** [comment]
> Can we move this into it's own workflow so we can path filer on breeze so we only run it when necessary?
> 
> Edit: Just saw that this is rather quick but with the gpu runner being on the more expensive side I think it would still make sense to only run this when relevant files are changed (velox/experimenta/breeze/* and maybe CMake/* because you use velox_resolve_dependency?)

Makes sense.  Please take a look at latest version of the PR where I try to do that.

File: .github/workflows/linux-build-base.yml (hunk 1 above)
Comment (dreveman) [reply .github/workflows/linux-build-base.yml]: Done

**Review by assignUser — APPROVED** [review]
Workflow looks good to me, thanks!

**Comment by assignUser** [comment]
Ah, one minor thing we could exclude breeze from the other builds trigger but that can be handled in a follow up.

**Comment by Yuhta** [comment]
@dreveman Yes the breeze tests should only be run when breeze itself is changed. The integration with wave can be run when wave is changed though.

**Review by Yuhta — APPROVED** [review]

**Comment by dreveman** [comment]
> @dreveman Yes the breeze tests should only be run when breeze itself is changed. The integration with wave can be run when wave is changed though.

Ok, I'll create a PR for only running breeze tests when changing breeze code. My only concern is that if wave is using breeze and we don't run wave tests when making breeze changes then it will be pretty easy to break that integration.

**Pull request closed without merging**